### PR TITLE
system tests: add assert(), and start using it

### PR DESCRIPTION
**Pull request by edsantiago**
Problem: the system test 'is()' checker was poorly thought out.
For example, there is no way to check for inequality or for
absence of a substring.

Solution, step 1: introduce new assert(), copied almost verbatim
from buildah, where it has been successful in addressing the
gaps in is().

The logical next step is to search the tests for 'die' and
for 'run', looking for negative assertions which we can
replace with assert(). There were a lot, and in the process
I found a number of ugly bugs in the tests themselves. I've
taken the liberty of fixing these.

Important note: at this time we have both assert() and is().
Replacing all instances of is() would be impossible to review.

Signed-off-by: Ed Santiago <santiago@redhat.com>

### CODE REVIEW

File: test/system/helpers.bash
Comment (edsantiago) [inline test/system/helpers.bash:809]: Unrelated change, simply a fix to make things cleaner.

File: test/system/030-run.bats
Comment (edsantiago) [inline test/system/030-run.bats:779]: YIKES. This was not only a cross-filesystem move (bad idea), but into a tempfs, so the original hosts file would be lost forever if the test crashed in the wrong place! My replacement is still scary, but a little less so.

File: test/system/065-cp.bats
Comment (edsantiago) [inline test/system/065-cp.bats:929]: Unnecessary: if the command failed, bats itself would abort immediately.

File: test/system/065-cp.bats
Comment (edsantiago) [inline test/system/065-cp.bats:933]: Completely brokey: `run` is regular bats, not my `run_podman` extension. It does not do the exit-status check. This was actually running the `1` command, which does not exist, and therefore it was failing, but nobody was checking status.

File: test/system/250-systemd.bats
Comment (edsantiago) [inline test/system/250-systemd.bats:73]: I have no idea what this was intended to do, but it's a NOP, so I've removed it.

**Review by edsantiago — COMMENTED** [review]
Suggestion for reviewers: start at the bottom (`helpers.bash`), look at `assert()`, then proceed with the rest.

A few inline comments intended to explain the reasons for less-obvious changes.

File: test/system/030-run.bats
Comment (Luap99) [reply test/system/030-run.bats:779]: Can we just not do that please, I run the test locally on my laptop. The test should never ever touch system configuration. This is a very bad idea if this crashes or another application on my system tries to use it while it is moved.

IMO such test should be done in the in container test where we have more control or the environment. Alternatively the test should create a new mount namespace where it would remove this file, however that would be super difficult to do properly.

Comment (edsantiago) [reply test/system/030-run.bats:779]: Yes, it is a stupendously bad idea. Thanks for the `$container` suggestion, I will do so tout de suite.

**Comment by edsantiago** [comment]
/hold

DO NOT MERGE YET. I've rebased on top of #13936 to test that the skip-e2e-tests logic works.

**Comment by rhatdan** [comment]
LGTM
@containers/podman-maintainers PTAL
@giuseppe @vrothberg @Luap99 @flouthoc PTAL

**Review by vrothberg — APPROVED** [review]
_Really_ nice work, @edsantiago!

/lgtm

**Comment by Luap99** [comment]
LGTM but we should fix https://github.com/containers/podman/pull/13935#discussion_r854443035 soon. This is a very bad idea.

**Comment by edsantiago** [comment]
>LGTM but we should fix https://github.com/containers/podman/pull/13935#discussion_r854443035 soon. This is a very bad idea

@Luap99 is the current implementation (running the test only in `$container` environment) not a good enough balance between mount namespace (hard) and risk (who cares what happens to a containerized test environment)?

**Comment by Luap99** [comment]
The current $container check never runs the test because there are no systems which run in a container AFAICT.
I am fine with the check but we have to make sure that it will actually run, otherwise we should remove the test.

**Pull request merged**
